random-svg-nft/contracts/EmojiNFT.sol

```diff
@@ -0,0 +1,136 @@
+
+  event RandomnessRequested(uint256 indexed requestId);
+
+  constructor(
+    address _vrfCoordinator,
+    bytes32 _keyHash,
+    uint64 _subscriptionId,
+    uint32 _callbackGasLimit,
+    uint32 _numWords,
+    uint16 _requestConfirmations
+  ) VRFConsumerBaseV2(_vrfCoordinator) ERC721("EmojiNFT", "EMOJI") {
+    vrfCoordinator = VRFCoordinatorV2Interface(_vrfCoordinator);
+    keyHash = _keyHash;
+    subscriptionId = _subscriptionId;
+    callbackGasLimit = _callbackGasLimit;
+    numWords = _numWords;
+    requestConfirmations = _requestConfirmations;
+  }
+
+  function mint() public returns (uint256 requestId) {
+    requestId = vrfCoordinator.requestRandomWords(
+      keyHash,
+      subscriptionId,
+      requestConfirmations,
+      callbackGasLimit,
+      numWords
+    );
+
+    requestToSender[requestId] = msg.sender;
+
+    emit RandomnessRequested(requestId);
+  }
+
+  function pickRandomColor(uint256 firstRandomNumber, uint256 secondRandomNumber, uint256 thirdRandomNumber)
+    internal
+    pure
+    returns (string memory)
+  {
+    uint256 r = firstRandomNumber % 256;
+    uint256 g = secondRandomNumber % 256;
+    uint256 b = thirdRandomNumber % 256;
+
+    return
+      string(
+        abi.encodePacked(
+          "rgb(",
+          Strings.toString(r),
+          ", ",
+          Strings.toString(g),
+          ", ",
+          Strings.toString(b),
+          ");"
+        )
+      );
+  }
+
+  function createOnChainSvg(string memory emoji, string memory color) internal pure returns(string memory svg) {
+    string memory baseSvg = "<svg xmlns='http://www.w3.org/2000/svg' preserveAspectRatio='xMinYMin meet' viewBox='0 0 350 350'><style>.base { font-size: 100px; }</style><rect width='100%' height='100%' style='fill:";
+    string memory afterColorSvg = "' /><text x='50%' y='50%' class='base' dominant-baseline='middle' text-anchor='middle'>";
+
+    svg = string(abi.encodePacked(baseSvg, color, afterColorSvg, emoji, "</text></svg>"));
+  }
+
+  function createTokenUri(string memory emoji, string memory svg) internal pure returns(string memory tokenUri) {
+    string memory json = Base64.encode(
+      bytes(
+        string(
+          abi.encodePacked(
+            '{"name": "',
+            emoji,
+            '", "description": "Random Emoji NFT Collection Powered by Chainlink VRF", "image": "data:image/svg+xml;base64,',
+            Base64.encode(bytes(svg)),
+            '"}'
+          )
+        )
+      )
+    );
+
+    tokenUri = string(
+      abi.encodePacked("data:application/json;base64,", json)
+    );
+  }
+
+  function fulfillRandomWords(uint256 requestId, uint256[] memory randomNumbers)
+    internal
+    override
+  {
+    uint256 tokenId = tokenIds.current();
+
+    uint256 emojiIndex = (randomNumbers[0] % emojis.length) + 1;
+    string memory emoji = emojis[emojiIndex];
+    string memory color = pickRandomColor(randomNumbers[1], randomNumbers[2], randomNumbers[3]);
```

Comment:
We are expecting 4 numbers here but above, line 44 we are allowing the constructor to determine the number of words, do we want to allow the ability to determine the number of words when we have to have at least 4?

Reply:
Good catch!